src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings({"rawtypes"})
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private int capacity;
+    private int size;
+    private Node[] nodes;
+
+    public MyHashMap() {
+        nodes = new Node[INITIAL_CAPACITY];
+        capacity = INITIAL_CAPACITY;
+    }
 
+    @SuppressWarnings({"unchecked"})
     @Override
     public void put(K key, V value) {
+        if (size >= capacity * LOAD_FACTOR) {
+            capacity = grow();
+        }
+        int index = indexCalculating(key);
+        Node<K, V> node = new Node<>(key, value);
+        if (nodes[index] == null) {
+            nodes[index] = node;
+        } else {
+            Node<K, V> nextNode = nodes[index];
+            if (checkDuplicate(node, nextNode)) {
+                return;
+            }
+            while (nextNode.next != null) {
+                nextNode = nextNode.next;
+                if (checkDuplicate(node, nextNode)) {
+                    return;
+                }
+            }
+            nextNode.next = node;
+        }
+        size++;
+    }
+
+    private int indexCalculating(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % capacity);
+    }
 
+    private boolean checkDuplicate(Node<K, V> node, Node<K, V> nextNode) {
+        if (Objects.equals(node.key, nextNode.key)) {
+            nextNode.value = node.value;
+            return true;
+        }
+        return false;
     }
 
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private int grow() {
+        capacity = capacity * 2;
+        Node[] elements = nodes;
```

Comment:
```suggestion
        Node<K, V>[] elements = nodes;
```
same in all places like that